File: Modules/Articles/Article/ArticleForm.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Dashboard.Articles.Forms;

[FormScript("Articles.Article")]
[BasedOnRow(typeof(ArticleRow), CheckNames = true)]
public class ArticleForm
{
    [Tab("Article")]
    public string Title { get; set; }
    public string HtmlContent { get; set; }
    public string Image { get; set; }

    [HideOnInsert]
    public int AutherId { get; set; }
    public bool IsActive { get; set; }
    public int CategoryId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    [Tab("Comments"), IgnoreName, CommentsGrid, LabelWidth("0")]
    public string CommentsGrid { get; set; } // column typd doesn't matter as it will be shown as Grid anyway
}

File: Modules/ServerTypes/Articles/ArticleForm.ts
import { StringEditor, HtmlContentEditor, ImageUploadEditor, LookupEditor, BooleanEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { CategoryDialog } from "@/Articles/Category/CategoryDialog";
import { CommentsGrid } from "@/Articles/Comments/CommentsGrid";
import { initFormType } from "@serenity-is/corelib/q";

export interface ArticleForm {
    Title: StringEditor;
    HtmlContent: HtmlContentEditor;
    Image: ImageUploadEditor;
    AutherId: LookupEditor;
    IsActive: BooleanEditor;
    CategoryId: LookupEditor;
    CreatedAt: DateEditor;
    UpdatedAt: DateEditor;
    CommentsGrid: CommentsGrid;
}

export class ArticleForm extends PrefixedContext {
    static formKey = 'Articles.Article';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ArticleForm.init)  {
            ArticleForm.init = true;

            var w0 = StringEditor;
            var w1 = HtmlContentEditor;
            var w2 = ImageUploadEditor;
            var w3 = LookupEditor;
            var w4 = BooleanEditor;
            var w5 = DateEditor;
            var w6 = CommentsGrid;

            initFormType(ArticleForm, [
                'Title', w0,
                'HtmlContent', w1,
                'Image', w2,
                'AutherId', w3,
                'IsActive', w4,
                'CategoryId', w3,
                'CreatedAt', w5,
                'UpdatedAt', w5,
                'CommentsGrid', w6
            ]);
        }
    }
}

[CategoryDialog]; // referenced types